packages/components/tests/integration/components/hds/tooltip/tooltip-button-test.js

```diff
@@ -0,0 +1,103 @@
+      return new Promise((resolve) => {
+        setTimeout(resolve, timeout);
+      });
+    }
+
+    await render(
+      hbs`<Hds::TooltipButton @text="More info." id="test-tooltip-button">info</Hds::TooltipButton>`
+    );
+
+    // Test that tooltip does not display by default:
+    assert.dom('.tippy-box').doesNotExist();
+
+    // Focus button to trigger tooltip display:
+    await focus('#test-tooltip-button');
+    assert.dom('.tippy-box').exists();
+
+    // Trigger escape key to close the tooltip:
+    await triggerKeyEvent('#test-tooltip-button', 'keydown', escapeKey);
+    await wait();
+    // test that the tooltip is now gone:
+    assert.dom('.tippy-box').doesNotExist();
+  });
+
+  test('the tooltip has a role of "tooltip"', async function (assert) {
+    await render(
+      hbs`<Hds::TooltipButton @text="More info." id="test-tooltip-button">info</Hds::TooltipButton>`
+    );
+    await focus('#test-tooltip-button');
+    assert.dom('.tippy-box').hasAttribute('role', 'tooltip');
+  });
+
+  test('it renders plain text content passed into the tooltip', async function (assert) {
+    await render(
+      hbs`<Hds::TooltipButton @text="More info." id="test-tooltip-button">info</Hds::TooltipButton>`
+    );
+    await focus('#test-tooltip-button');
+    assert.dom('.tippy-content').hasText('More info.');
+  });
+
+  test('it renders rich HTML and text content passed into the tooltip', async function (assert) {
+    await render(
+      hbs`<Hds::TooltipButton @text="<em>em</em> <strong>strong</strong>" id="test-tooltip-button">info</Hds::TooltipButton>`
+    );
+    await focus('#test-tooltip-button');
+    assert.dom('.tippy-content em').exists().hasText('em');
+    assert.dom('.tippy-content strong').exists().hasText('strong');
+  });
+
+  test('the button has an aria-describedby attribute with a value matching the tooltip id', async function (assert) {
+    await render(
+      hbs`<Hds::TooltipButton @text="Hello" data-test-tooltip-button>info</Hds::TooltipButton>`
+    );
+    await focus('[data-test-tooltip-button]');
+    assert.dom('[data-test-tooltip-button]').hasAttribute('aria-describedby');
+    assert.dom('[data-tippy-root]').hasAttribute('id');
+
+    assert.strictEqual(
+      this.element
+        .querySelector('[data-test-tooltip-button]')
+        .getAttribute('aria-describedby'),
+      this.element.querySelector('[data-tippy-root]').getAttribute('id')
+    );
+  });
+
+  // ATTRIBUTES
+
+  test('it should spread all the attributes passed to the component on the element', async function (assert) {
+    await render(
+      hbs`
+      <Hds::TooltipButton @text="Here is more info." id="test-tooltip-button" class="my-class" data-test1 data-test2="test">info</Hds::TooltipButton>
+      `
+    );
+    assert.dom('#test-tooltip-button').hasClass('my-class');
+    assert.dom('#test-tooltip-button').hasAttribute('data-test1');
+    assert.dom('#test-tooltip-button').hasAttribute('data-test2', 'test');
+  });
+});
```

Comment:
Are there more tests I should add for the tooltipButton component?

I can't figure out how to properly test the "placement" and "offset" options that come from Tippy.js. It uses an inline style with transform translate3d to implement position the tooltip relative to the screen so there aren't expected values I could check for or classnames or anything predictable it seems.

Reply:
I would say this functionality lives in `tippy.js` and it's tested there we only make use of their API.
However, if you'd like to test it, there is a `assert.dom('.tippy-box').hasStyle({top: '1px'})` that may work for this.

Reply:
Thanks, it uses an inline transform translate3d style for the positioning so the values aren't easily predictable so I think I will leave as is since we are relying on the Tippy.js API vs. adding this option ourselves.